File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.cpp
class Solution {
public:
    int countGoodSubstrings(string s) {
        size_t n = s.length();

        if(n < 3) return 0;

        short int count = 0;

        for(int i=0; i<n-2; i++)
        if(s[i] != s[i+1] && s[i+1] != s[i+2] && s[i+2] != s[i]) count++;

        return int(count);
    }
};

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.go
func countGoodSubstrings(s string) int {
    var n int = len(s)
    if n < 0 { return 0 }

    var count int = 0
    for i:=0; i<n-2; i++ {
        if s[i] != s[i+1] && s[i] != s[i+2] && s[i+1] != s[i+2] { count++ }
    }

    return count
}